### PR TITLE
Fix for Issue -63 add VCN (subnets) Logging - Issue #63

### PR DESCRIPTION
**Pull request by syedthameem85**
# Proposed change

### Automated testing

If you're running automated testing for this module, we would love to hear from you, and potentially integrate it to the module standard workflow.

### Manual testing

If no automated testing is run, please ensure that at least the three steps below are passing without any error.

- [ ] Running `terraform apply` on each example provided with this module provisions the intended resource(s) without any errors.
- [ ] Modifying module's *Input Variables* after initial provisioning behaves as intended, i.e: any updateable properties are ameneded without recreation of the resource(s).
- [ ] Running `terraform destroy` on each example provided with this module destroys all the resources created by this module and only the resources created by this module.

## Checklist before submitting your PR

- [ ] My code follows [the style guidelines of this project](../tree/main/docs/codingconventions.adoc)
- [ ] these changes provision new resources:
  - [ ] I have updated the README introduction section (README.adoc)
  - [ ] I have updated the README introduction section (README.md)
- [ ] these changes adds any new variables:
  - [ ] I have updated [docs/terraformoptions.adoc](../tree/main/docs/terraformoptions.adoc) to include each variable
- [ ] I have updated the changelog to include an entry for these changes
- [ ] I have updated all provided examples, including each README file and all applicable code blocks
- [ ] these changes generates no new warnings
- [ ] Any dependent changes have been merged and published in upstream modules

Note: *If you are not an Oracle employee, to contribute to an Oracle-sponsored open-source project, you need to sign the [Oracle Contributor Agreement (OCA)](https://oca.opensource.oracle.com/).*

### CODE REVIEW

**Comment by syedthameem85** [comment]
@hyder  - This PR is the solution for the issue https://github.com/oracle-terraform-modules/terraform-oci-vcn/issues/63 - https://github.com/oracle-terraform-modules/terraform-oci-vcn/issues/63

File: variables.tf [hunk 1]
```diff
@@ -171,4 +171,16 @@ variable "subnets" {
...
   type        = string
+}
+
+variable "log_retention_duration" {
+  type        = number
+  default     = 30
+  description = "Log retention duration"
+}
+
+variable "enable_vcn_logging" {
```
Comment (hyder) [inline variables.tf:182]: bring the control variable of the feature before the variable used to configure the feature i.e. define enable_ first and then the retention.

File: variables.tf [hunk 2]
```diff
@@ -171,4 +171,16 @@ variable "subnets" {
 variable "tenancy_id" {
   description = "Tenancy OCID"
   type        = string
+}
+
+variable "log_retention_duration" {
+  type        = number
+  default     = 30
```
Comment (hyder) [inline variables.tf:178]: please explicitly indicate the duration.

File: vcn.tf [hunk 3]
```diff
@@ -35,3 +35,29 @@ module "subnet" {
   count = length(var.subnets) > 0 ? 1 : 0
 
 }
+
+locals {
+  vcn_id = oci_core_vcn.vcn.id
+  subnet = {
+    for key, value in var.subnets : key => contains(keys(value), "name") ? value.name : key
+  }
+  service_logdef = { for k in local.subnet : format("%s_%s", k, "log") => { loggroup = "loggrp", service = "flowlogs", resource = k } }
```
Comment (hyder) [inline vcn.tf:44]: please format your code before committing

File: vcn_gateways.tf [hunk 4]
```diff
@@ -174,14 +174,23 @@ resource "oci_core_route_table" "service_gw" {
 ###################
 # NAT Gateway (NGW)
 ###################
+resource "oci_core_public_ip" "nat_gateway_public_ip" {
```
Comment (hyder) [inline vcn_gateways.tf:177]: please submit separately so we can track. Don't mix features/fixes.

**Review by hyder — CHANGES_REQUESTED** [review]
please see comments.

**Comment by hyder** [comment]
Why did you close it?

**Pull request closed without merging**